portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,23 +40,31 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int maxNumComments = 1;
     Query query = new Query(COMMENT_TABLE_NAME ).addSort(TIMESTAMP_COLUMN_NAME, SortDirection.ASCENDING);
-    PreparedQuery results = datastore.prepare(query);
+    PreparedQuery results = dataStore.prepare(query);
     List<String> myComments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-        myComments.add((String) entity.getProperty(COMMENT_COLUMN_NAME));
-    }
-    String commentJSON = convertToJson(myComments);
+    String maxNumCommentsParam = request.getParameter("max-num");
+    if (maxNumCommentsParam != null && !maxNumCommentsParam.isEmpty()) {
+        maxNumComments = Integer.parseInt(maxNumCommentsParam);
```

Comment:
Are you intending `maxNumComments` to be the default max? What could happen if a client specifies `max-num=fizzbuzz`?

Reply:
yes! I am intending for maxNumComments = 1 to be the default max. A client cannot specify that value because the input tag is set to only accept numbers :)

Reply:
Is there more I can do to further improve on this problem/piece of code?

Reply:
> A client cannot specify that value because the input tag is set to only accept numbers :)

If they're using the web page it will be hard to do, but a different client might not make an HTTP request directly from the command line or some such... In which case you might end up with a `NumberFormatException`.

Some options:
* Use a strongly typed message in the request (which is out of scope for this PR)
* catch the `NumberFormatException` and throw an `BadRequestException` back to the client.
* Validate the input independently at the start of the message.

I'm fine with leaving this as is, but just wanted to point it out :)

Reply:
sounds good! I will make a separate PR addressing this point :)